src/main/java/com/istb/app/services/profesor/ProfesorServiceImpl.java

```diff
@@ -69,19 +79,19 @@ public Profesor findByCedula(String cedula) {
 	@Override
 	public Map<String, String> update(Profesor profesor, int id) {
 		Map<String, String> errorAttributes = new HashMap<>();
-		
+
 		Profesor _profesor = profesorRepository.findById(id).orElse(null);
-		
+
 		Profesor existeCorreo = profesorRepository.findByCorreoAndIdIsNot(profesor.getCorreo(), id);
```

Comment:
Aqui podrias utilizar un boolean, adicionalmente para que no trates con valores `null` podrias utilizar un objeto `Optional` de tipo `Profesor` en el método del repositorio.

```suggestion
		boolean existeCorreo = profesorRepository.findByCorreoAndIdIsNot(
		    profesor.getCorreo(), id ).get().isPresent();
```
El cambio en [ProfesorService](https://github.com/cess15/grupo-1-control-asistencias/blob/feature-email-verify/src/main/java/com/istb/app/repository/ProfesorRepository.java) seria:
`Optional<Profesor> findByCorreoAndIdIsNot(String correo, int id);`